### PR TITLE
NDFrame.filter -> NDFrame.select

### LINKED ISSUE #26642: Rename NDFrame.filter to .select?
I've made a PR to remove ``NDFrame.select``, see #26641. That method was deprecated in 0.21.

There was a discussion in #12401, where many participants agreed that removing the old ``select`` and having ``NDFrame.filter`` renamed to ``NDFrame.select`` would be a better naming scheme. I agree on that for several of the reasons mentioned in the thread.

What do people think of such a name change and a related deprecation of the name (not functionality) ``.filter``?

### PR DESCRIPTION
**Pull request by topper-123**
- [x] closes #26642
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

This PR moves ``NDFrame.filter`` over to ``NDFrame.select`` and deprecates ``NDFrame.filter``.

I'm not completely happy with the siganture from ``.filter``, and got some discussion points in #26642. The current PR can be seen as a first version, to be changed, if there's agreement,

### CODE REVIEW

**Review by jreback — CHANGES_REQUESTED** [review]
we should have a discussion about these semantics in the issue, can you summarize there

**Comment by TomAugspurger** [comment]
We need to have at least one release without NDFrame.select, before we can change it to be the new name for filter, right?

**Comment by topper-123** [comment]
I've set up an issue at #26642.

**Comment by topper-123** [comment]
I'm closing this donw temporarily. As I'vr said in #26642, I'm in favor ofmaking a specialized method that wouldn't overlap too much ``.loc``.
